### PR TITLE
Improve comment of ExpectedValue in db stress

### PR DESCRIPTION
**Pull request by hx235**
**Context/Summary:**
https://github.com/facebook/rocksdb/pull/11424 made me realize there are a couple gaps in my `ExpectedValue` comments so I updated them, along with separating `ExpectedValue` into separate files so it's clearer that `ExpectedValue` can be used without updating `ExpectedState` (e.g, TestMultiGet() where we care about value base of expected value but not updating the ExpectedState).

**Test:**
CI

### CODE REVIEW

**Comment by ajkr** [comment]
Is it a common pattern to put declarations and definitions into differently named files? MustHaveNotExisted() would be in expected_state.cc and expected_value.h according to this PR. I know we split db_impl.h into multiple implementation files, but do we ever split one implementation file into multiple header files? I'd prefer to not do the latter if possible.

**Comment by hx235** [comment]
> Is it a common pattern to put declarations and definitions into differently named files? MustHaveNotExisted() would be in expected_state.cc and expected_value.h according to this PR. I know we split db_impl.h into multiple implementation files, but do we ever split one implementation file into multiple header files? I'd prefer to not do the latter if possible.

Thanks!! Fixed.

File: db_stress_tool/expected_value.h [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  const ExpectedValue final_value_;
+};
+
+// `ExpectedValueHelper` provides utils to parse `ExpectedValue` to obtain
+// useful info about it in db stress
+class ExpectedValueHelper {
+ public:
+  // Return whether value is expected not to exist from begining till the end
+  // of the read based on `pre_read_expected_value` and
+  // `pre_read_expected_value`.
```
Comment (jowlyzhang) [inline db_stress_tool/expected_value.h:182]: nit: pre_read_expected_value \ post_read_expected_value. Similar nit for below `MustHaveNotExisted`.

In "value is expected not to exist ...", what does "value" refer to?

**Review by jowlyzhang — COMMENTED** [review]
Thanks for the PR. LGTM, just some minor comments.

File: db_stress_tool/expected_value.h (hunk 1 above)
Comment (hx235) [reply db_stress_tool/expected_value.h:182]: Good question ....

Comment (hx235) [reply db_stress_tool/expected_value.h]: Fixed

**Review by jowlyzhang — APPROVED** [review]

**Pull request closed without merging**